### ISSUE TITLE
Outline not populating (Add more log messages to outline logic)

### ISSUE THREAD

**Issue by TiemenSch** [James-Yu/LaTeX-Workshop#389]
### Description

Hi James,

I'm sorry to say so, but I fear the outline has stopped working for me again. 
Right now I can't even get it to populate with all latex in document.tex (e.g. without input files). Tested it with:

```latex
\documentclass{article}

\begin{document}
\section{Hello}
world.
\section{Allo}
Allo.
\end{document}
```
### Log messages

<details>
<summary>LaTeX Workshop Output</summary>

[14:40:40] File watcher: responding to change in d:\Tiemen\Documents\test\document.tex
[14:40:40] Parsing d:\Tiemen\Documents\test\document.tex
[14:40:40] Linter for root file started.
[14:40:40] Linter for root file running command chktex with arguments -wall,-n8,-n22,-n30,-n35,-n46,-e16,-q,-f%f:%l:%c:%d:%k:%n:%m
,d:\Tiemen\Documents\test\document.tex
[14:40:40] BUILD command invoked.
[14:40:40] Found root file from active editor: d:\Tiemen\Documents\test\document.tex
[14:40:40] Root file remains unchanged from: d:\Tiemen\Documents\test\document.tex.
[14:40:40] Building root file: d:\Tiemen\Documents\test\document.tex
[14:40:40] Build root file d:\Tiemen\Documents\test\document.tex
[14:40:40] Toolchain step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,d:/Tiemen/Documents/test/document
[14:40:40] Parsing d:\Tiemen\Documents\test\document.tex for outline
[14:40:40] Parsing d:\Tiemen\Documents\test\document.tex for outline
[14:40:40] Parsing d:\Tiemen\Documents\test\document.tex for outline
[14:40:40] Linter for active file started.
[14:40:40] Linter for active file running command chktex with arguments -wall,-n8,-n22,-n30,-n35,-n46,-e16,-q,-I0,-f%f:%l:%c:%d:%k:%n:%m

[14:40:40] Linter for root file successfully finished in 0s 386ms
[14:40:40] Linter log parsed with 0 messages.
[14:40:40] Linter for active file successfully finished in 0s 260ms
[14:40:40] Linter log parsed with 0 messages.
[14:40:40] Formatted d:\Tiemen\Documents\test\document.tex
[14:40:41] LaTeX log parsed with 0 messages.
[14:40:41] Toolchain of length 1 finished.
[14:40:41] Successfully built d:\Tiemen\Documents\test\document.tex
[14:40:41] Refresh PDF viewer for d:\Tiemen\Documents\test\document.pdf
[14:40:41] Preview PDF file: d:\Tiemen\Documents\test\document.pdf
[14:41:00] Root file remains unchanged from: d:\Tiemen\Documents\test\document.tex.
</details>

### Additional Information

As I understand it, you currently parse the .tex files yourself, scouting for commands, right? Wouldn't it be more efficient to let LaTeX do as much work for you as possible? 

My guess would be that it would be more robust to rely on **synctex** and the **aux** file.
I think the chapter/section titles are already summarized in the .aux file, and code locations are already stored in synctex's file, albeit a tad bit more obscure there.

If a user has the **hyperref** package enabled, it's even easier, as it just outputs plain section names in an **.out** file:
<details>
<summary>hyperref .out file</summary>

>  \BOOKMARK [1][-]{section.1}{Introduction}{}% 1
>  \BOOKMARK [1][-]{section.2}{Background}{}% 2
>  \BOOKMARK [2][-]{subsection.2.1}{Stuffing}{section.2}% 3
>  \BOOKMARK [2][-]{subsection.2.2}{Cookies}{section.2}% 4
>  \BOOKMARK [2][-]{subsection.2.3}{Ovens}{section.2}% 5 
>  \BOOKMARK [1][-]{section.3}{Case study}{}% 6
>  \BOOKMARK [2][-]{subsection.3.1}{Recipe}{section.3}% 7
>  \BOOKMARK [2][-]{subsection.3.2}{Tasting}{section.3}% 8

</details>

**Comment by James-Yu** [comment]
This is what I got.

![image](https://user-images.githubusercontent.com/4210342/35420904-139afa2c-027b-11e8-95a0-aa6e56732e03.png)

**Comment by James-Yu** [comment]
As to `aux` file, I have to say the main problem is inconsistency. When `hyperref` is included the syntax is completely different, albeit 'looks' similar.

**Comment by caofan** [comment]
When switching tex files, both of which have `\begin{document}`, the outline is not updated with the active tex file.

**Comment by TiemenSch** [comment]
### Minimal example
Today I retried the simple example and the outline is shown now -- strange enough. My main concern is the robustness of the outline. Is it possible to include it in every build or save? There are file watchers in place, but as an end-user the outline seems to be populating/updating rather randomly. It would be comforting/predictable to have an option to force it to update, for example on every save (just like you can toggle building it in settings) or just in your commands list.

Also, is it logged anywhere why it would leave the outline empty? That would help a lot in identifying the actual issue.

### Aux
I haven't inspected the aux files that much, but rather saw the necessary info ending up there. I do see the same syntax popping up in my more complicated projects (with hyperref), too, however. 

Complicated stuff:
``\@writefile{toc}{\defcounter {refsection}{0}\relax }\@writefile{toc}{\contentsline {section}{\numberline {2}Background}{3}{section.2}}``

Overlap with simple stuff:
``\@writefile{toc}{\contentsline {section}{\numberline {1}Hello}{1}}``

I may be overlooking variations, but it seems hopeful? From ``\@writefile{toc}`` to the section name seems to include all info you need.

**Comment by TiemenSch** [comment]
Update and more info:
Previously in #346 you fixed indentations of ``\section`` commands and such. 
That is broken for me now. Latest VS Code, 3.12.3 Latex-Workshop.

Forcing it to update is possible, by heading to a random file like the ".fls" by latexmk and heading back to the document.tex main file. However, this seems to be unexpected behaviour. 

Would it be wise to open up a separate issue for the "persistence" of the outline in an active Latex-Workshop project? It already has a "main file" registered somewhere and it would give the most consistent experience if it just always updates that on save or build (based on a setting?). Perhaps even regardless of the file that is open?

**Comment by James-Yu** [comment]
The outline is actually rebuild upon save, and everything is working fine on my side. I'm afraid that working examples are still required to find out what was wrong.

I do sometimes observe that the whole outline disappear when opening a `tex` file, and returning from other formats will let it show. But I would suppose this is a vscode bug since in any time `tex` files should have the outline section appear, with or without contents.

**Comment by TiemenSch** [comment]
Would it be possible to make the outline parsing slightly more verbose?

**Comment by James-Yu** [comment]
I may add a toggle on this later.

**Comment by jhintsch** [comment]
Hi @James-Yu,

nice plugin, I have switched to your plugin + vscode from TexnicCenter.
Is the use of \input actually supported right now for the Latex Outline side window?

Best,
Johannes

**Comment by TiemenSch** [comment]
I saw that there was an update to the outline logic in the changelog, so I decided to fiddle with some basic examples again. Sadly, on my system, the outline is still very unstable. Also on the times that I got it to populate, clicking items did not move my cursor to the sections, but only to the right file. 

Do you have examples that it absolutely should work for or at least does so on your system?

**Comment by giahung24** [comment]
Just a little out-of-topic, but I have a problem with outline when using \include{chapter}. 
In this simple example, it can't detect chapter in included file.
![image](https://user-images.githubusercontent.com/11850677/40004549-39961066-5796-11e8-8fce-01476734435d.png)

In my principal project, where each chapter is imported from a separated file, the outline just shows 2 random chapter titles.

**Comment by giahung24** [comment]
@James-Yu I don't know it you have fixed it the last update (don't see in changelog), but my problem above is gone. The outline is populated perfectly now.
Thanks.

**Comment by TiemenSch** [comment]
I can confirm a populated outline too :)! Thanks a bunch!

**Comment by jlelong** [comment]
It was fixed in 2ccae14 while working on an other issue.
